zanata-model/src/main/java/org/zanata/model/validator/UniqueValidator.java

```diff
@@ -62,7 +62,7 @@ private int countRows(Object value) {
                 session.getSessionFactory().getClassMetadata(value.getClass());
         String idName = metadata.getIdentifierPropertyName();
         // FIXME was EntityMode.POJO
-        Serializable id = metadata.getIdentifier(value);
+        Serializable id = metadata.getIdentifier(value, null);
```

Comment:
Perhaps you should cast the null to a particular type, so that a future overload of `getIdentifier` doesn't break compilation.